src/test/java/org/apache/sysml/test/integration/functions/paramserv/SparkDataPartitionerTest.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sysml.test.integration.functions.paramserv;
+
+import static org.apache.sysml.hops.OptimizerUtils.DEFAULT_BLOCKSIZE;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.api.java.function.Function2;
+import org.apache.sysml.api.DMLScript;
+import org.apache.sysml.parser.Statement;
+import org.apache.sysml.runtime.controlprogram.context.SparkExecutionContext;
+import org.apache.sysml.runtime.controlprogram.paramserv.ParamservUtils;
+import org.apache.sysml.runtime.controlprogram.paramserv.spark.DataPartitionerSparkMapper;
+import org.apache.sysml.runtime.controlprogram.paramserv.spark.DataPartitionerSparkReducer;
+import org.apache.sysml.runtime.matrix.data.MatrixBlock;
+import org.apache.sysml.runtime.matrix.data.MatrixIndexes;
+import org.apache.sysml.runtime.util.DataConverter;
+import org.junit.Assert;
+import org.junit.Test;
+
+import scala.Tuple2;
+
+public class SparkDataPartitionerTest {
+
+	@Test
+	public void testSparkDataPartitionerDC() {
+		DataPartitionerSparkMapper mapper = new DataPartitionerSparkMapper(Statement.PSScheme.DISJOINT_CONTIGUOUS, 3);
+		Map<Integer, Tuple2<MatrixBlock, MatrixBlock>> result = doPartitioning(mapper);
+
+		Assert.assertEquals(3, result.size());
+		result.values().forEach(v -> {
+			Assert.assertEquals(v._1.getNumRows(), v._2.getNumRows());
+			double[] features = v._1.getDenseBlockValues();
+			double[] labels = v._2.getDenseBlockValues();
+			for (int i = 0; i < v._1.getNumRows(); i++) {
+				for (int j = 0; j < v._1.getNumColumns(); j++) {
+					Assert.assertEquals((int)labels[i], (int)features[i * v._1.getNumColumns() + j] / v._1.getNumColumns());
+				}
+			}
+		});
+	}
+
+	@Test
+	public void testSparkDataPartitionerDR() {
```

Comment:
Please compare with local DR partitioner to check for consistent global ordering.